RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,830 @@
+
+There are a few things we need to consider for this to be secure as there are possible scenarios that can reduce the security
+in the atomic swap. 
+
+When looking at the two lock heights, the first lock height should be sufficiently large enough to give ample time for Alice to post the Tari UTXO transaction and for it to be mined with a safe number of confirmations,
+and for Bob to post the Monero transaction and for it to be mined with a safe number of confirmations. The second lock
+height should give ample time for Alice after the first lock height to re-claim her Tari. Larger heights here might make
+refunds slower, but it should be safer in giving more time to finalize this. 
+
+Allowing both to claim the Tari after the second lock height is, on face value, a safer option. This can be done by enabling
+either party to claim the script with the lapse transaction. The counterparty can then claim the Monero. However, this
+will open up an attack vector to enable either party to claim the Monero while claiming the Tari. Either party could trivially
+pull off such a scheme by performing a front-running attack and having a bit of luck. The counterparty monitors all broadcast 
+transactions to base nodes. Upon identifying the lapse transaction, they do two things; in quick succession, broadcast 
+their lapse transaction and the transaction to claim the Monero, both with sufficiently high fees. Base nodes will 
+prefer to mine transactions with the higher fees, and thus the counterparty can walk away with both the Tari and the 
+Monero.
+
+It is also possible to prevent the transaction from being mined after being submitted to the mempool. This can be caused by a combination
+of a too busy network, not enough fees, or a too-small period in the time locks. When one of these atomic swap transactions gets published to a mempool, we effectively already have all the details exposed. For the atomic swaps, it means we already revealed the Monero key, although
+the actual Tari transaction has not been mined. But this is true for any HTLC or like script on any blockchain. But in the odd
+chance that this does happen whereby the fees are too little and time locks not enough, it should be possible to do a child-pays-for-parent
+transaction to bump up the fees on the transaction to get it mined and confirmed.
+
+## Key construction
+
+Using [multi-signatures](https://reyify.com/blog/flipping-the-scriptless-script-on-schnorr) with Schnorr signatures, we
+need to ensure that the keys are constructed so that key cancellation attacks are not possible. To do this, we create new
+keys from the chosen public keys \\(K_a'\\) and \\(K_b'\\)
+
+$$
+\begin{aligned}
+K_a &=  \hash{\hash{K_a' \cat K_b'} \cat K_a' } * K_a' \\\\
+k_a &=  \hash{\hash{K_a' \cat K_b'} \cat K_a' } * k_a' \\\\
+K_b &=  \hash{\hash{K_a' \cat K_b'} \cat K_b' } * K_b' \\\\
+k_b &=  \hash{\hash{K_a' \cat K_b'} \cat K_b' } * k_b' \\\\
+\end{aligned}
+\tag{1}
+$$
+
```

Comment:
I think a section here called "**Key equivalence across different elliptic curves**" or the like is warranted, similar to how "**Key construction**" is explained.
This is essentially a Discrete log equivalence (DLEQ) proof and should be mentioned. Reference to the Monero article <https://web.getmonero.org/resources/research-lab/pubs/MRL-0010.pdf> and more general use for example <https://projekter.aau.dk/projekter/files/260345149/Electronic_voting_application_based_on_public_verifiable_secret_sharing.pdf> paragraph 5.3.1 and  <https://crypto.stackexchange.com/questions/60127/key-equivalence-across-different-elliptic-curves> should be made, possible with a very brief one sentence summary about the differences between the two approaches.